### PR TITLE
Fixed RL evaluations

### PR DESCRIPTION
**Pull request by kim-mskw**
Fixed RL evaluations to use compare_and_save_policies function instead of comparing in 

- enabled the origignal function compare_and_save_policies()
- was basically redundant in code
- also changed the collect initial experience to be overwritten to 1 when it is 0 because otherwise the buffer gives an error, because it has no entries to sample from

### CODE REVIEW

File: assume/reinforcement_learning/learning_role.py [hunk 1]
```diff
@@ -232,8 +237,14 @@ def compare_and_save_policies(self) -> None:
...
+            Curretnly the best evaluation metrik is still assessed by the development team and preliminary we use the avergae rewards.
         """
-        modes = ["reward", "profit", "regret"]
+        # modes = ["reward", "profit", "regret"]
+        modes = ["reward"]
+
+        # add current rewrad to list of all rewards
+        self.rl_eval_rewards.append(current_avg_reward)
+
         for mode in modes:
```
Comment (nick-harder) [inline assume/reinforcement_learning/learning_role.py:248]: do we even need this for loop now if we are using only a single mode?

File: assume/reinforcement_learning/learning_role.py [hunk 2]
```diff
@@ -243,26 +254,27 @@ def compare_and_save_policies(self) -> None:
...
                 and self.rl_eval_regrets[-1] != 0
             ):
-                self.max_eval_regret = self.rl_eval_regrets[-1]
-                dir_name = "lowest_regret"
-                value = self.max_eval_regret
+                # self.max_eval_regret = self.rl_eval_regrets[-1]
+                # dir_name = "lowest_regret"
+                # value = self.max_eval_regret
+
+                return NotImplementedError("Evaluation Strategie is not implemnted yet")
```
Comment (nick-harder) [inline assume/reinforcement_learning/learning_role.py:274]:
```suggestion
                return NotImplementedError("Evaluation Strategy is not implemented yet")
```

**Review by nick-harder — CHANGES_REQUESTED** [review]
I didn't fully get it, so are we now saving when each individual reward is max or the average among all agents? what did we decided from out last long discussion on this topic?

File: assume/reinforcement_learning/learning_role.py (hunk 1 above)
Comment (kim-mskw) [reply assume/reinforcement_learning/learning_role.py:248]: I inteded to keep the variability for the future but make it as easy as possible and still have the average reward as the target. But Folrian and I made it more generic and still have the avg reward.

**Comment by kim-mskw** [comment]
> get it, so are we now saving when each individual reward is max or the average among all agents? what did we decided from out last long discussion on this topic?

we still did it with the average among all agents since we did not decide on an alternative. With Florians upcoming commit it is programmed in such a way, that we can easily change that in the future though.

**Pull request merged**